cmake/API.cmake

```diff
@@ -138,367 +138,294 @@ function(add_fprime_subdirectory FP_SOURCE_DIR)
-# files. Thus, only the SOURCE_FILE variable needs to be set and then the register call can be made.
-# This is the only required lines in a module CMakeLists.txt.
-#
-# ```
-# set(SOURCE_FILE
-#     MyComponent.fpp
-#     SomeFile.cpp
-#     MyComponentImpl.cpp)
-#
-# register_fprime_module()
-# ```
+####
+function(fprime_attach_custom_targets BUILD_TARGET_NAME)
+    setup_module_targets("${BUILD_TARGET_NAME}")
+endfunction()
+
+####
+# Function `register_fprime_library`:
 #
-# ### Non-Autocoded and Autocode-Only Modules Example ###
+# Registers a library using the fprime build system. This comes with dependency management and fprime
+# autocoding capabilities. The first argument is the name of this module and will become the build target
+# name. Sources, autocoder inputs, link dependencies, and headers are each passed in after the directives
+# SOURCES, AUTOCODER_INPUTS, DEPENDS, and HEADERS, respectively.  Each directive may be used one time and
+# dictates the contents of arguments until the next directive.
 #
-# Modules that do not require autocoding need not specify *.xml files as source. Thus, code-only modules just define
-# *.cpp. **Note:** dependency inference is only done when autocoder inputs (.fpp, .xml) are supplied.
+# **Example:**
 #
 # ```
-# set(SOURCE_FILE
-#     SomeFile1.cpp
-#     Another2.cpp)
-#
-# register_fprime_module()
+# register_fprime_library(
+#         MyFprimeModule
+#     SOURCES
+#         source1.cpp
+#         source2.cpp
+#     AUTOCODER_INPUTS
+#         model.fpp
+#     DEPENDS
+#         -lm
+#     HEADERS
+#         module.h
+# )
 # ```
-# Modules requiring only autocoding may just specify *.xml files.
 #
-# ```
-# set(SOURCE_FILE
-#     MyComponent.fpp)
+# > [!NOTE]  
+# > This delegates to CMake's `add_library` call. The library argument EXCLUDE_FROM_ALL is supported.
 #
-# register_fprime_module()
-# ```
+# **MODULE_NAME**: (optional) module name. Default: ${FPRIME_CURRENT_MODULE}
+# **ARGN**: sources, autocoder inputs, etc preceded by a directive (i.e. SOURCES or DEPENDS)
 #
-# ### Specific Dependencies and Linking in Modules Example ###
+####
+function(register_fprime_library)
+    fprime_add_library_build_target(${ARGN})
+    clear_historical_variables()
+    # Set up target/ targets for this module
+    fprime_attach_custom_targets("${INTERNAL_MODULE_NAME}")
+endfunction(register_fprime_library)
+
+####
+# Function `register_fprime_module`:
 #
-# Some modules need to pick a specific set of dependencies and link flags. This can be done
-# with the `MOD_DEPS` variable. This feature can be used to pick specific implementations
-# for some fprime modules that implement to a generic interface like the console logger implementation.
-# 
-# ```
-# set(SOURCE_FILE
-#     MyComponent.fpp
-#     SomeFile.cpp
-#     MyComponentImpl.cpp)
+# See `register_fprime_library`. This provides the same capability as `register_fprime_library` using the
```

Comment:
Add warning DEPRECATED, at least in the docstring ?

Reply:
Let's do this when we break folder targets from module targets.